src/test/HST/Util/MessagesTests.hs

```diff
@@ -15,56 +17,110 @@ testMessages = describe "HST.Util.Messages" $ do
-    str
-      `shouldBe` unlines [ firstFile ++ ":1:8-1:21:"
-                         , "1 | module ExampleQueue1 where"
-                         , "           ^^^^^^^^^^^^^"
-                         ]
-  it "should correctly display a multi-line SrcSpan with startY < endY" $ do
-    str <- runM . runInputFile $ displayCodeExcerpt $ Just multiLineSrcSpan1
-    str
-      `shouldBe` unlines
-      [ firstFile ++ ":5:1-6:11:"
-      , "    vvvvvvvvvvvvvvvv"
-      , "5 | empty :: Queue a"
-      , "6 | empty = []"
-      , "    ^^^^^^^^^^"
-      ]
+  it "should return the empty string if no source span is given"
+    $ let str = run . runInputFileNoIO exampleMap $ displayCodeExcerpt Nothing
+      in str `shouldBe` []
+  it "should display an error message if end line < start line"
+    $ let str = run . runInputFileNoIO exampleMap
+            $ displayCodeExcerpt
+            $ Just invalidSrcSpan2
+      in str `shouldBe` "The source span 2:4-1:5 of `"
+         ++ exampleFilePath
+         ++ "` cannot be fully displayed!"
+  it "should display an error message if end column < start column and start line = end line"
+    $ let str = run . runInputFileNoIO exampleMap
+            $ displayCodeExcerpt
+            $ Just invalidSrcSpan
+      in str `shouldBe` "The source span 1:4-1:3 of `"
+         ++ exampleFilePath
+         ++ "` cannot be fully displayed!"
+  it "should display an error message if end line < start line"
+    $ let str = run . runInputFileNoIO exampleMap
+            $ displayCodeExcerpt
+            $ Just invalidSrcSpan2
+      in str `shouldBe` "The source span 2:4-1:5 of `"
+         ++ exampleFilePath
+         ++ "` cannot be fully displayed!"
+  it "should correctly display a one-line excerpt"
+    $ let str = run . runInputFileNoIO exampleMap
+            $ displayCodeExcerpt
+            $ Just oneLineSrcSpan
+      in str
+         `shouldBe` unlines [ exampleFilePath ++ ":1:8-1:21:"
+                            , "1 | module ExampleQueue1 where"
+                            , "           ^^^^^^^^^^^^^"
+                            ]
+  it "should correctly display a multi-line SrcSpan with end column < start column"
+    $ let str = run . runInputFileNoIO exampleMap
+            $ displayCodeExcerpt
+            $ Just multiLineSrcSpan2
+      in str
+         `shouldBe` unlines
+         [ exampleFilePath ++ ":5:10-6:6:"
+         , "             vvvvvvv"
+         , "5 | empty :: Queue a"
+         , "6 | empty = []"
+         , "    ^^^^^"
+         ]
+  it "should correctly display a multi-line SrcSpan with start column < end column"
+    $ let str = run . runInputFileNoIO exampleMap
+            $ displayCodeExcerpt
+            $ Just multiLineSrcSpan1
+      in str
+         `shouldBe` unlines
+         [ exampleFilePath ++ ":5:1-6:11:"
+         , "    vvvvvvvvvvvvvvvv"
+         , "5 | empty :: Queue a"
+         , "6 | empty = []"
+         , "    ^^^^^^^^^^"
+         ]
```

Comment:
These tests should be switched, so that `multiLineSrcSpan2` comes after `multiLineSrcSpan1`. Additionally, you could change one of these tests or add a new one to test the shortened excerpts which are produced when the source span spans more than five lines.